src/main/java/com/google/sps/servlets/Utils.java

```diff
@@ -86,6 +90,36 @@ static Calendar loadCalendarClient() throws IOException {
     return new Calendar.Builder(HTTP_TRANSPORT, JSON_FACTORY, credential).build();
   }
 
+  // Query datastore with the current users email to retrieve user info entity
+  static Entity getCurrentUserEntity() {
+    UserService userService = UserServiceFactory.getUserService();
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+    String userEmail = userService.getCurrentUser().getEmail();
+    Query query = new Query("UserInfo")
+        .setFilter(new Query.FilterPredicate("email", Query.FilterOperator.EQUAL, userEmail));
+    PreparedQuery results = datastore.prepare(query);
+    return results.asSingleEntity();
+  }
+
+  // Retrieve the family id from the user info and fetch their family entity from datastore
+  static Entity getCurrentFamilyEntity(Entity userInfoEntity) {
```

Comment:
If you're expecting getCurrentUserEntity and getCurrentFamilyEntity to be called together, consider creating a single method that does the lookup for both user entity and family.

Reply:
Ok, I think for now I will leave them as is